index.bs

```diff
@@ -13244,6 +13269,165 @@ the exact steps to take if <dfn>an exception was thrown</dfn>, or by explicitly
 </div>
 
 
+<h3 id="synthetic-module-records">Synthetic module records</h3>
+
+A <dfn export>Synthetic Module Record</dfn> is used to represent information about a module that is
+defined by specifications.
+The set of exported names is static, and determined at creation time (as an argument to
+[$CreateSyntheticModule$]), while the set of exported values can be changed over time using
+[$SetSyntheticModuleExport$].
+It has no imports or dependencies.
+
+Note: A [=Synthetic Module Record=] could be used for defining a variety of module types: for
+example, built-in modules, or JSON modules, or CSS modules.
+
+Note: [=Synthetic Module Records=] are being developed in concert with the authors of the
+<cite>JavaScript Standard Library</cite> proposal, and might eventually move to the ECMAScript
+specification. [[JSSTDLIB]] [[ECMA-262]].
+
+In addition to the [=Module Record Fields=], Synthetic Module Records have the additional fields
+listed below.
+Each of these fields is initially set in [$CreateSyntheticModule$].
+
+<table id="table-synthetic-module-record-fields" class="data">
+<caption>Additional Fields of Synthetic Module Records</caption>
+<thead>
+  <tr>
+    <th>Field Name
+    <th>Value Type
+    <th>Meaning
+<tbody>
+  <tr>
+    <td>\[[ExportNames]]
+    <td>List of String
+    <td>A List of all names that are exported.
+  <tr>
+    <td>\[[EvaluationSteps]]
+    <td>An abstract operation
+    <td>An abstract operation that will be performed upon evaluation of the module, taking the
+        [=Synthetic Module Record=] as its sole argument.
+        These will usually set up the exported values, by using [$SetSyntheticModuleExport$].
+        They must not modify \[[ExportNames]].
+        They may return an abrupt completion.
+</table>
+
+<h4 id="createsyntheticmodule">CreateSyntheticModule</h4>
+
+<div algorithm>
+The abstract operation <dfn abstract-op>CreateSyntheticModule</dfn>(|exportNames|, |evaluationSteps|, |realm|, |hostDefined|) creates a [=Synthetic Module Record=] based upon
```

Comment:
Since we are in WebIDL, I wonder if Create SyntheticModule and SetSyntheticModuleExport would be more clean if phrased as worded algorithms, e.g., "To <dfn>create a synthetic module record</dfn>...". HostDefined also seems redundant here when the steps can close over outer variables (as they can in web specs, but not JS).

Reply:
I'd prefer to keep this close to ecma-speak for now, in case we decide to move it back under the TC39 umbrella.